Update error when using server-ref for primitive

### Describe the bug

When using a server ref that has been declared with:
```ts
export default {
  plugins: [
    ServerRef({
      state: {
        foo: 'bar'
      }
    })
  ]
}
```
Then imported with:
```ts
import foo from 'server-ref:foo'
```
When we try to update the ref in a browser instance, there is an error in the other browser instances: `ReferenceError: payload is not defined`.

Also I tried to reproduce directly in the repo to add an example in the example app, but I also have an error when trying to start the app:
```
 [ERROR] Failed to resolve entry for package "vite-plugin-vue-server-ref". The package may have incorrect main/module/exports specified in its package.json. [plugin externalize-deps]

 [...]

    vite.config.ts:3:22:
      3 │ import ServerRef from 'vite-plugin-vue-server-ref'
        ╵                       ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Reproduction

https://github.com/tonai/vue-app

### System Info

```Shell
System:
    OS: Linux 5.19 Ubuntu 22.04.2 LTS 22.04.2 LTS (Jammy Jellyfish)
    CPU: (8) x64 11th Gen Intel(R) Core(TM) i5-1145G7 @ 2.60GHz
    Memory: 22.06 GB / 30.74 GB
    Container: Yes
    Shell: 5.1.16 - /bin/bash
  Binaries:
    Node: 18.12.1 - ~/.nvm/versions/node/v18.12.1/bin/node
    npm: 8.19.2 - ~/.nvm/versions/node/v18.12.1/bin/npm
  Browsers:
    Chrome: 113.0.5672.126
    Firefox: 115.0
```


### Used Package Manager

pnpm

### Validations

- [X] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [X] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

--- a/src/generate.ts
+++ b/src/generate.ts
@@ -70,7 +70,7 @@ if (import.meta.hot) {
   function applySet(newData) {
     skipWatch = true
     onSet.forEach(fn => fn(newData))
-    ${type === 'ref' ? 'data.value = payload.data' : 'reactiveSet(data, newData)'}
+    ${type === 'ref' ? 'data.value = newData' : 'reactiveSet(data, newData)'}
     ${debug ? `console.log("[server-ref] [${key}] incoming", newData)` : ''}
     ${diff ? 'makeClone()' : ''}
     skipWatch = false

--- a/test/output/generate-ref-foo-bar.js
+++ b/test/output/generate-ref-foo-bar.js
@@ -33,7 +33,7 @@ if (import.meta.hot) {
   function applySet(newData) {
     skipWatch = true
     onSet.forEach(fn => fn(newData))
-    data.value = payload.data
+    data.value = newData
     skipWatch = false
   }
   import.meta.hot.on("vue-server-ref", (payload) => {

--- a/test/output/generate-ref-foo.js
+++ b/test/output/generate-ref-foo.js
@@ -33,7 +33,7 @@ if (import.meta.hot) {
   function applySet(newData) {
     skipWatch = true
     onSet.forEach(fn => fn(newData))
-    data.value = payload.data
+    data.value = newData
     skipWatch = false
   }
   import.meta.hot.on("vue-server-ref", (payload) => {

